File: Codeforces/1374E1.py
n, k = [int(x) for x in input().split()]

both = []
bob = []
alice = []

for i in range(n):
    t, a, b = [int(x) for x in input().split()]
    if(a==1 and b==1):
        both.append(t)
    elif(a==1):
        alice.append(t)
    elif(b==1):
        bob.append(t)

both.sort()
bob.sort()
alice.sort()
y = max(k-min(len(bob),len(alice)),0)

soma_both=0
sa = 0
sb = 0

poss = (len(both)+min(len(alice),len(bob))>=k)
if(poss):
    for i in range(y):
        soma_both+=both[i]

    for i in range(max(k-y,0)):
        sa += alice[i]

    for i in range(max(k-y,0)):
        sb += bob[i]

    ptr1 = k-y
    ptr2 = k-y

    resposta = sa+sb+soma_both
    for i in range(y,len(both)):
        ptr1-=1
        ptr2-=1
        if(ptr2==-1 or ptr1==-1):
            break
        soma_both+=both[i]
        sa-=alice[ptr1]
        sb-=bob[ptr2]
        if(soma_both+sa+sb>=0):
            resposta = min(resposta,soma_both+sa+sb)
    print(resposta)
else:
    print(-1)

File: Codeforces/1374E1.cpp
#include<bits/stdc++.h>
using namespace std;

int main(){
    int n, k, t, a, b;
    vector<int> both, alice, bob;

    cin >> n >> k;
    for(int i=0;i<n;i++){
        cin >> t >> a >> b;
        if(a&&b){
            both.push_back(t);
        }
        else if(a){
            alice.push_back(t);
        }
        else if(b){
            bob.push_back(t);
        }
    }

    sort(both.begin(),both.end());
    sort(bob.begin(),bob.end());
    sort(alice.begin(),alice.end());    
    bool poss = ((both.size()+min(bob.size(),alice.size())>=k));


    if(poss){
        int ptr = min(min((int)bob.size(),(int)alice.size()),k);
        int lim = min((int)both.size(),k);
        int sb=0, sa=0, acm=0, answ, ptr2 = k-ptr;
        for(int i=0;i<ptr;i++){
            sb+=bob[i];
        }
        for(int i=0;i<ptr;i++){
            sa+=alice[i];
        }
        for(int i=0;i<ptr2;i++){
            acm+=both[i];
        }
        answ = acm+sb+sa;
        for(int i=ptr2;i<lim;i++){
            sb-=bob[--ptr];
            sa-=alice[ptr];
            acm+=both[ptr2++];
            answ = min(answ,acm+sb+sa);
        }
        cout << answ << '\n';
    }
    else{
        cout << -1 << '\n';
    }

    return 0;
}